py/obj.c

```diff
@@ -159,28 +159,25 @@ bool mp_obj_is_callable(mp_obj_t o_in) {
 // Furthermore, from the v3.4.2 code for object.c: "Practical amendments: If rich
 // comparison returns NotImplemented, == and != are decided by comparing the object
 // pointer."
-bool mp_obj_equal(mp_obj_t o1, mp_obj_t o2) {
-    // Float (and complex) NaN is never equal to anything, not even itself,
-    // so we must have a special check here to cover those cases.
-    if (o1 == o2
-        #if MICROPY_PY_BUILTINS_FLOAT
-        && !mp_obj_is_float(o1)
-        #endif
-        #if MICROPY_PY_BUILTINS_COMPLEX
-        && !mp_obj_is_type(o1, &mp_type_complex)
-        #endif
-        ) {
-        return true;
+mp_obj_t mp_obj_equal_bop(mp_obj_t o1, mp_obj_t o2, bool not_equal) {
```

Comment:
I tried changing this function signature to:
```c
mp_obj_t mp_binary_op_equal(mp_binary_op_t op, mp_obj_t o1, mp_ob_t o2) {
    mp_obj_t local_true = (op == MP_BINARY_OP_NOT_EQUAL) ? mp_const_false : mp_const_true;
    ...
}
```
and it made the code size smaller by about 16 bytes on almost all ports.  And IMO it's a bit neater this way because it mirrors `mp_binary_op()`, just a special version for equal/not equal.

Reply:
Makes sense. I'll make that change.